### PR TITLE
Feature/business crud

### PR DESCRIPTION
**Pull request by Elinos**
Simple CRU(no Delete for now) for Businesses

### CODE REVIEW

**Comment by Elinos** [comment]
[CRUD for Businesses](https://trello.com/c/r8nwZtsT/27-crud-for-businesses)

File: WebApp.API/Controllers/BusinessController.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        [HttpGet]
+        [Produces("application/json")]
+        public async Task<IActionResult> GetAsync()
+        {
+            List<BusinessResponse> businesses = await _businessService.GetAllAsync();
+            return Ok(businesses);
+        }
+
+        [HttpGet]
+        [Route("{id}", Name = "GetBusinessById")]
```
Comment (ivopashov) [inline WebApp.API/Controllers/BusinessController.cs]: what route does this generate - `business/{id}` ?

File: WebApp.API/Controllers/BusinessController.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        [Produces("application/json")]
+        public async Task<IActionResult> GetByIdAsync(int id)
```
Comment (ivopashov) [inline WebApp.API/Controllers/BusinessController.cs]: I think you can simplify the name to mere `Get` - it is clear that it is by id by the param.

File: WebApp.API/Controllers/BusinessController.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        [Produces("application/json")]
+        public async Task<IActionResult> CreateAsync([FromBody] BusinessRequest model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            BusinessResponse newBusiness = await _businessService.CreateAsync(model);
+            return CreatedAtRoute("GetBusinessById", new { id = newBusiness.Id }, newBusiness);
```
Comment (ivopashov) [inline WebApp.API/Controllers/BusinessController.cs]: just curious - what is this method - `CreatedAtRoute `?

File: WebApp.API/Services/BusinessConverter.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using WebApp.API.Models.Db;
+using WebApp.API.Models.Input;
+using WebApp.API.Models.Output;
+
+namespace WebApp.API.Services
+{
+    public class BusinessConverter : IBusinessConverter
```
Comment (ivopashov) [inline WebApp.API/Services/BusinessConverter.cs]: Can we use auto mapper or something similar?

File: WebApp.API/Controllers/BusinessController.cs (hunk 1 above)
Comment (Elinos) [inline WebApp.API/Controllers/BusinessController.cs]: Yes

File: WebApp.API/Controllers/BusinessController.cs (hunk 3 above)
Comment (Elinos) [inline WebApp.API/Controllers/BusinessController.cs]: It returns 201(created) and also adds location header for better discoverability.

File: WebApp.API/Data/Migrations/20161203170430_businesses.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+using Microsoft.EntityFrameworkCore.Migrations;
+using WebApp.API.Data;
+
+namespace WebApp.API.Data.Migrations
+{
+    [DbContext(typeof(ApplicationDbContext))]
+    [Migration("20161203170430_businesses")]
+    partial class businesses
```
Comment (Elinos) [inline WebApp.API/Data/Migrations/20161203170430_businesses.Designer.cs]: This is auto-generated migration

File: WebApp.API/Services/BusinessConverter.cs (hunk 4 above)
Comment (Elinos) [inline WebApp.API/Services/BusinessConverter.cs]: Not sure what is supported in .Net Core and also AutoMapper sometimes creates more problems than benefits, so I wanted to keep it simple for now and make the mapping by myself.

**Pull request merged**